src/Symfony/Component/Messenger/Middleware/HandleMessageMiddleware.php

```diff
@@ -41,7 +42,7 @@ public function handle(Envelope $envelope, StackInterface $stack): Envelope
         $handler = null;
         $message = $envelope->getMessage();
         foreach ($this->handlersLocator->getHandlers($envelope) as $handler) {
-            $handler($message);
+            $envelope = $envelope->with(HandledStamp::fromCallable($handler, $handler($message)));
```

Comment:
isnt there also an alias to  use as key here? I.e. `HandledStamp::getHandlerAlias()`

Reply:
No, at least not with the fwb. But I hesitated making the `HandlersLocator` consistent with the `SenderLocator`, so providing an alias would be possible (but no usage in core)

Reply:
ok. i kinda assumed handlers/senders locator would/should follow the same design yes.

Reply:
Introduced handler alias as well along handler description :)